<launch>

    <arg name="start_pt" default= '-7.43, -15.3, 0.8978652, 0'/>
    <arg name="use_rviz" default="true"/>
    <arg name="use_matplotlib" default="false"/>
    <arg name="is_sim" default="true"/>
    <arg name="map_file" default="$(find svea_core)/resources/maps/floor2.yaml" />    


    <arg name="symps" default="$(find av09)/src/symps.txt" />
    <arg name="faults" default="$(find av09)/src/faults.txt" />

    <node pkg="av09" type="simulation_node.py" name="simulation_node" output="screen">
        
    </node>
    <node pkg="av09" type="detection_node.py" name="detection_node" output="screen">
        
    </node>
    <node pkg="av09" type="diagnose_plan_node.py" name="diagnose_plan_node" output="screen">
	<param name="start_pt" value="$(arg start_pt)"/>

    </node>
    <node pkg="av09" type="vehicle_to_firestore.py" name="vehicle_to_firestore" output="screen">
        
    </node>


    

    <group if="$(eval is_sim == true)">
        <node type="rviz" name="rviz" pkg="rviz" args="-d $(find svea_core)/resources/rviz/SVEA_floor2.rviz" />
    </group>

    <node pkg="map_server" type="map_server" name="map_server" output="screen" args="$(arg map_file)" />
    <rosparam command="load" file="$(find svea_core)/resources/params/obstacles.yaml" />
    <rosparam command="load" file="$(find svea_core)/resources/params/track.yaml" />

    <node pkg="av09" type="actuation_node.py" name="actuation_node" output="screen">
        <param name="start_pt" value="$(arg start_pt)"/>
        <param name="use_rviz" value="$(arg use_rviz)"/>
        <param name="use_matplotlib" value="$(arg use_matplotlib)"/>
        <param name="is_sim" value="$(arg is_sim)"/>
    </node>


</launch>
